server/src/test/java/org/opensearch/gateway/remote/RemoteClusterStateServiceTests.java

```diff
@@ -545,6 +575,292 @@ public void testWriteIncrementalMetadataSuccess() throws IOException {
+        );
+        assertEquals("Diff manifest null which is required for downloading cluster state", error.getMessage());
+    }
+
+    public void testGetClusterStateUsingDiff_NoDiff() throws IOException {
+        ClusterStateDiffManifest diffManifest = ClusterStateDiffManifest.builder().build();
+        ClusterState clusterState = generateClusterStateWithOneIndex().nodes(nodesWithLocalNodeClusterManager()).build();
+        ClusterMetadataManifest manifest = ClusterMetadataManifest.builder()
+            .diffManifest(diffManifest)
+            .stateUUID(clusterState.stateUUID())
+            .stateVersion(clusterState.version())
+            .metadataVersion(clusterState.metadata().version())
+            .clusterUUID(clusterState.getMetadata().clusterUUID())
+            .routingTableVersion(clusterState.routingTable().version())
+            .build();
+        ClusterState updatedClusterState = remoteClusterStateService.getClusterStateUsingDiff(manifest, clusterState, "test-node");
+        assertEquals(clusterState.getClusterName(), updatedClusterState.getClusterName());
+        assertEquals(clusterState.metadata().clusterUUID(), updatedClusterState.metadata().clusterUUID());
+        assertEquals(clusterState.metadata().version(), updatedClusterState.metadata().version());
+        assertEquals(clusterState.metadata().coordinationMetadata(), updatedClusterState.metadata().coordinationMetadata());
+        assertEquals(clusterState.metadata().getIndices(), updatedClusterState.metadata().getIndices());
+        assertEquals(clusterState.metadata().templates(), updatedClusterState.metadata().templates());
+        assertEquals(clusterState.metadata().persistentSettings(), updatedClusterState.metadata().persistentSettings());
+        assertEquals(clusterState.metadata().transientSettings(), updatedClusterState.metadata().transientSettings());
+        assertEquals(clusterState.metadata().getCustoms(), updatedClusterState.metadata().getCustoms());
+        assertEquals(clusterState.metadata().hashesOfConsistentSettings(), updatedClusterState.metadata().hashesOfConsistentSettings());
+        assertEquals(clusterState.getCustoms(), updatedClusterState.getCustoms());
+        assertEquals(clusterState.stateUUID(), updatedClusterState.stateUUID());
+        assertEquals(clusterState.version(), updatedClusterState.version());
+        assertEquals(clusterState.getRoutingTable().version(), updatedClusterState.getRoutingTable().version());
+        assertEquals(clusterState.getRoutingTable().getIndicesRouting(), updatedClusterState.getRoutingTable().getIndicesRouting());
+        assertEquals(clusterState.getNodes(), updatedClusterState.getNodes());
+        assertEquals(clusterState.getBlocks(), updatedClusterState.getBlocks());
+    }
+
+    public void testGetClusterStateUsingDiff() throws IOException {
+        ClusterState clusterState = generateClusterStateWithOneIndex().nodes(nodesWithLocalNodeClusterManager()).build();
```

Comment:
Overall the tests look good but `generateClusterStateWithOneIndex` should generate state with some value for all objects like templates, so that we can get more coverage for diff application.